src/cudadecoder/cuda-decoder.cc

```diff
@@ -299,6 +300,10 @@ CudaDecoder::~CudaDecoder() {
   // Stopping h2h tasks
   h2h_threads_running_ = false;
   n_h2h_main_task_todo_cv_.notify_all();
+  for(auto& t : *cpu_dedicated_threads_) {
```

Comment:
space after (.  I don't see why it was necessary to make the variable a pointer though.  It would be freed anyway in the destructor.  (maybe the join() matters though).

Reply:
It is all about lifetime. As a normal member variable, it's guaranteed to be deconstructed in the reverse declaration order, at the moment, those threads might still be running. With a raw pointer, you have more control over it.

Reply:
OK, my mis-understanding. Master have the fix, close it.